lib/npm/tagger.js

```diff
@@ -98,3 +100,28 @@ Tagger.prototype.list = function (pkg, callback) {
     callback(null, pack.distTags);
   });
 };
+
+/**
+ * Async/await wrapper for running a tag add but deleting if it fails
+ * @function wrap
+ * @param {Object} opts Options
+ *  @param {Boolean} opts.tag Should we run the tag operation or not
+ *  @param {Object} opts.spec Spec for the operation
+ *  @param {String} opts.version Semver version for the given package
+ * @param {Function} fn Function to await for given operation
+ * @returns {Promise} via async fn
+ */
+Tagger.prototype.wrap = async function ({ tag, spec, version }, fn) {
+  const { name: pkg, env } = spec;
+
+  this.log.info('tagger.wrap', { tag, spec, version });
+  if (tag) await thenify(this, 'add', { pkg, version, tag: env  });
+
+  try {
+    return await fn();
```

Comment:
This patterns just reads like a typical callback-style function. Why use an `async` function here when the rest of the prototype (`.list`, `.add`, etc) are all callback based?

Reply:
@SivanMehta I put it on this prototype rather than on the build-manager itself since it seemed to belong here conceptually. It does put it into an in between state since the whole tagger itself is still callback based but i didn't want to keep rabbit holing on refactoring everything all in one shot since I already took too long with it.